### ISSUE TITLE
Factor out common code

### ISSUE THREAD

**Issue by thomas-fossati** [veraison/corim#11]
Factor out common code in a `corim/common` package

_Originally posted by @thomas-fossati in https://github.com/veraison/corim/pull/7#discussion_r710075095_

**Comment by Sukuna0007Abhi** [comment]
I started working on it, I can complete it by tomorrow, there are some more packages  common pattern implementation , almost  got 40+ files updates, will raise PR, thanks pls assign in your free time sir @thomas-fossati @yogeshbdeshpande

**Comment by Sukuna0007Abhi** [comment]
Yes @thisisharsh7 It's a big PR,  big chunk of codes are added, so it will take some time to complete it, I am only focusing on it now if you want,  you can work on this issue #76 no one will work on it, the guy who messaged on that issue is currently working on another issue so let's go for it

**Comment by Sukuna0007Abhi** [comment]
Or you can try for #155  that's also I thought of working that's also big so you can go for this also, if you want
